TypeCobol/Compiler/Nodes/Data.cs

```diff
@@ -282,31 +283,187 @@ public virtual DataType PrimitiveDataType
             }
         }
 
-
-        private int? _length = null;
         /// <summary>
         /// TODO This method should be split like this:
         /// - PhysicalLength 
         /// - PhysicalLengthWithChildren
-        /// - LogicalLength
-        /// - LogicalLengthWithChildren
         /// </summary>
-        public virtual int Length
+        private long _physicalLength = 0;
+        public virtual long PhysicalLength
+        {
+            get
+            {
+                if (_physicalLength > 0)
+                {
+                    return _physicalLength;
+                }
+
+                if (children != null)
+                {
+                    if(Picture != null || (Usage != null && Usage != DataUsage.None && Children.Count == 0))
+                    {
+                        _physicalLength = ((DataDefinitionEntry) CodeElement).PhysicalLength;
+                    }
+                    else
+                    {
+                        foreach (var node in children)
+                        {
+                            var dataDefinition = (DataDefinition)node;
+
+                            _physicalLength += dataDefinition.PhysicalLength;
+                            _physicalLength += dataDefinition.SlackBytes;
+
+                            if (dataDefinition is DataRedefines)
+                            {
+                                SymbolReference redefined = ((DataRedefinesEntry)dataDefinition.CodeElement)?.RedefinesDataName;
+                                var results = SymbolTable.GetVariables(redefined);
```

Comment:
I say a mistake in my previous code review, the correct method to use is SymbolTable.GetRedefinedVariable